src/codegen/compile.ml

```diff
@@ -8407,15 +8533,38 @@ module Persistence = struct
     load_stable_actor env ^^
     compile_test I64Op.Eqz ^^
     (E.if1 I64Type
-      begin
-        OldStabilization.old_destabilize env actor_type (NewStableMemory.upgrade_version env)
-      end
-      begin
-        recover_actor env actor_type ^^
-        NewStableMemory.restore env
-      end) ^^
-    StableMem.region_init env ^^
+      (empty_actor env actor_type)
+      (recover_actor env actor_type)
+    ) ^^
+    NewStableMemory.restore env ^^
     UpgradeStatistics.add_instructions env
+
+  let initialize env actor_type =
+    register_stable_type env actor_type
+end (* EnhancedOrthogonalPersistence *)
+
+(* As fallback when doing persistent memory layout changes. *)
+module GraphCopyStabilization = struct
+  let is_graph_stabilization_started env =
+    E.call_import env "rts" "is_graph_stabilization_started" ^^ Bool.from_rts_int32
+
+  let start_graph_stabilization env actor_type =
+    EnhancedOrthogonalPersistence.create_type_descriptor env actor_type ^^
+    E.call_import env "rts" "start_graph_stabilization"
+
+  let graph_stabilization_increment env =
+    E.call_import env "rts" "graph_stabilization_increment" ^^ Bool.from_rts_int32
+
+  let start_graph_destabilization env actor_type complete_initialization =
```

Comment:
`complete_initialization` is unused?

Reply:
Thanks. This is a leftover that can be deleted. I moved the completion logic to  `destabilization_increment`.